### PR TITLE
Conduit support

### PR DESCRIPTION
**Pull request by swiftyspiffy**
Adds new methods for Helix/EventSub:
- GetConduits
- CreateConduit
- UpdateConduit
- DeleteConduit
- GetConduitShards
- UpdateConduitShards

Tested:
- Creating conduits
- Getting conduits
- Deleting conduits
- Updating conduit with 2 shards, one for webhook and one for websocket
- Getting conduit shard statuses

### CODE REVIEW

**Review by Mahsaap — APPROVED** [review]

File: TwitchLib.Api.Helix/EventSub.cs [hunk 1]
```diff
@@ -132,5 +137,125 @@ public async Task<bool> DeleteEventSubSubscriptionAsync(string id, string client
...
+        /// Creates a new conduit.
+        /// </summary>
+        /// <param name="request">Request body parameters identifying conduit details</param>
+        /// <param name="clientId">optional Client ID to override the use of the stored one in the TwitchAPI instance</param>
+        /// <param name="accessToken">optional access token to override the use of the stored one in the TwitchAPI instance</param>
+        /// <returns cref="CreateConduitsResponse">Returns a list of your conduits.</returns>
+        public async Task<CreateConduitsResponse> CreateConduits(CreateConduitsRequest request, string clientId = null,
+            string accessToken = null)
+        {
+            if (request.ShardCount is <= 0 or > 20_0000)
```
Comment (AoshiW) [inline TwitchLib.Api.Helix/EventSub.cs:163]:
```suggestion
            if (request.ShardCount is <= 0 or > 20_000)
```

File: TwitchLib.Api.Helix/EventSub.cs [hunk 2]
```diff
@@ -132,5 +137,125 @@ public async Task<bool> DeleteEventSubSubscriptionAsync(string id, string client
...
+        /// Updates a conduit’s shard count. To delete shards, update the count to a lower number, and the shards above the count will be deleted. For example, if the existing shard count is 100, by resetting shard count to 50, shards 50-99 are disabled.
+        /// </summary>
+        /// <param name="request">Request body parameters identifying conduit details</param>
+        /// <param name="clientId">optional Client ID to override the use of the stored one in the TwitchAPI instance</param>
+        /// <param name="accessToken">optional access token to override the use of the stored one in the TwitchAPI instance</param>
+        /// <returns cref="UpdateConduitsResponse">Returns a list of your conduits.</returns>
+        public async Task<UpdateConduitsResponse> UpdateConduits(UpdateConduitsRequest request, string clientId = null,
+            string accessToken = null)
+        {
+            if (request.ShardCount is <= 0 or > 20_0000)
```
Comment (AoshiW) [inline TwitchLib.Api.Helix/EventSub.cs:180]:
```suggestion
            if (request.ShardCount is <= 0 or > 20_000)
```

**Review by AoshiW — CHANGES_REQUESTED** [review]

File: TwitchLib.Api.Helix/EventSub.cs (hunk 1 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix/EventSub.cs]: good catch

File: TwitchLib.Api.Helix/EventSub.cs (hunk 2 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix/EventSub.cs]: good catch

**Review by Mahsaap — APPROVED** [review]
Didn't even see that shard number in the docs. Good catch.

**Pull request merged**